packages/core/phase3/lib/pncUpdateRequestGenerators/remandGenerator/remandGenerator.test.ts

```diff
@@ -0,0 +1,217 @@
+          ThirdLevelCode: "00",
+          BottomLevelCode: "00",
+          OrganisationUnitCode: "B000000"
+        },
+        PNCDisposalType: 2051,
+        PNCAdjudicationExists: true,
+        ResultClass: ResultClass.ADJOURNMENT_POST_JUDGEMENT,
+        ResultQualifierVariable: [{ Code: "LE" }]
+      }
+    ] as Result[]
+  } as Offence
+
+  const pncUpdateDataset = generatePncUpdateDatasetFromOffenceList([offence])
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.ArrestSummonsNumber =
+    "1101ZD0100000410780J"
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing.DateOfHearing = new Date("2024-12-05")
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.RemandStatus = "CB"
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.BailConditions = [
+    "This is a dummy bail condition."
+  ]
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.ForceOwner = {
+    TopLevelCode: "A",
+    SecondLevelCode: "02",
+    ThirdLevelCode: "BJ",
+    BottomLevelCode: "01",
+    OrganisationUnitCode: "A02BJ01"
+  }
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing.CourtType = "MCA"
+  pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Hearing.CourtHouseName = "Magistrates' Courts London Croydon"
+
+  return pncUpdateDataset
+}
+
+const operation = {
+  code: "NEWREM",
+  data: {
+    nextHearingDate: new Date("2024-12-11T10:11:12.000Z"),
+    nextHearingLocation: {
+      TopLevelCode: "B",
+      SecondLevelCode: "00",
+      ThirdLevelCode: "00",
+      BottomLevelCode: "00",
+      OrganisationUnitCode: "B000000"
+    }
+  }
+} as Operation<PncOperation.REMAND>
+
+describe("remandGenerator", () => {
+  it("should generate remand request with all fields", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+
+    const result = remandGenerator(pncUpdateDataset, operation)
+
+    expect(result).toStrictEqual({
+      operation: "NEWREM",
+      request: {
+        croNumber: null,
+        forceStationCode: "02YZ",
+        pncCheckName: null,
+        pncIdentifier: null,
+        arrestSummonsNumber: "11/01ZD/01/410780J",
+        hearingDate: "05122024",
+        nextHearingDate: "11122024",
+        pncRemandStatus: "B",
+        remandLocationCourt: "",
+        psaCourtCode: "9998",
+        courtNameType1: "Magistrates' Courts London Croydon MCA",
+        courtNameType2: "Magistrates' Courts London Croydon MCA",
+        localAuthorityCode: "0000",
+        bailConditions: ["This is a dummy bail condition."]
+      }
+    })
+  })
+
+  it("should set bail conditions to empty array when PNC remand status is 'C'", () => {
+    const pncUpdateDataset = createPncUpdateDataset()
+    pncUpdateDataset.AnnotatedHearingOutcome.HearingOutcome.Case.HearingDefendant.RemandStatus = "PB"
+
+    const result = remandGenerator(pncUpdateDataset, operation)
+
+    expect(result).toStrictEqual({
```

Comment:
I think we just do:

```suggestion
    expect(result.bailConditions).toHaveLength(0)
```

as we have already tested all the fields in the previous test so it's just noise in my opinion unless there are some other attributes that change as a result of changing the remand status.